File: lib/src/state/escrow.ts
import * as borsh from "borsh";
import { 
    Buffer 
} from "buffer";
import { 
    PublicKey 
} from "@solana/web3.js";


export class Escrow {

    authority: PublicKey;
    mint: PublicKey;
    quantity: number;
    bump: number;

    constructor(props: {
        authority: PublicKey,
        mint: PublicKey,
        quantity: number,
        bump: number,
    }) {
        this.authority = props.authority;
        this.mint = props.mint;
        this.quantity = props.quantity;
        this.bump = props.bump;
    }

    toBuffer() { 
        return Buffer.from(borsh.serialize(EscrowSchema, this)) 
    }
    
    static fromBuffer(buffer: Buffer) {
        const escrow = borsh.deserialize(EscrowSchema, Escrow, buffer);
        return new Escrow({
            authority: escrow.authority,
            mint: escrow.mint,
            quantity: escrow.quantity,
            bump: escrow.bump,
        });
    }
}

export const EscrowSchema = new Map([
    [ Escrow, { 
        kind: 'struct', 
        fields: [ 
            ['authority', [32]],
            ['mint', [32]],
            ['quantity', 'u8'],
            ['bump', 'u8'],
        ],
    }]
]);

File: program/src/state/escrow.rs
use borsh::{ BorshDeserialize, BorshSerialize };
use solana_program::pubkey::Pubkey;

#[derive(BorshDeserialize, BorshSerialize)]
pub struct Escrow {
    pub authority: Pubkey,
    pub mint: Pubkey,
    pub quantity: u64,
    pub bump: u8,
}

impl Escrow {

    pub const SEED_PREFIX: &'static str = "escrow";
    
    pub fn new(
        authority: Pubkey,
        mint: Pubkey,
        quantity: u64,
        bump: u8,
    ) -> Self {

        Escrow {
            authority,
            mint,
            quantity,
            bump,
        }
    }
}